src/bun.js/bindings/JSStringDecoder.cpp

```diff
@@ -0,0 +1,342 @@
+    if (callFrame->argumentCount() < 1) {
+        RELEASE_AND_RETURN(throwScope, JSC::JSValue::encode(castedThis->end(vm, lexicalGlobalObject, nullptr)));
+    }
+
+    auto buffer = callFrame->uncheckedArgument(0);
+    JSC::JSUint8Array* view = JSC::jsDynamicCast<JSC::JSUint8Array*>(buffer);
+    if (!view) {
+        throwVMTypeError(lexicalGlobalObject, throwScope, "Expected Uint8Array"_s);
+        return JSValue::encode(jsUndefined());
+    }
+    RELEASE_AND_RETURN(throwScope, JSC::JSValue::encode(castedThis->end(vm, lexicalGlobalObject, view)));
+}
+
+static JSC_DECLARE_HOST_FUNCTION(jsStringDecoderPrototypeFunction_write);
+static JSC_DEFINE_HOST_FUNCTION(jsStringDecoderPrototypeFunction_write,
+    (JSC::JSGlobalObject * globalObject, JSC::CallFrame* callFrame))
+{
+    return IDLOperation<JSStringDecoder>::call<jsStringDecoderPrototypeFunction_writeBody>(*globalObject, *callFrame, "write");
+}
+static JSC_DECLARE_HOST_FUNCTION(jsStringDecoderPrototypeFunction_end);
+static JSC_DEFINE_HOST_FUNCTION(jsStringDecoderPrototypeFunction_end,
+    (JSC::JSGlobalObject * globalObject, JSC::CallFrame* callFrame))
+{
+    return IDLOperation<JSStringDecoder>::call<jsStringDecoderPrototypeFunction_endBody>(*globalObject, *callFrame, "end");
+}
+
+/* Hash table for prototype */
+static const HashTableValue JSStringDecoderPrototypeTableValues[]
+    = {
+          { "write"_s, static_cast<unsigned>(JSC::PropertyAttribute::Function), NoIntrinsic, { HashTableValue::NativeFunctionType, jsStringDecoderPrototypeFunction_write, 1 } },
+          { "end"_s, static_cast<unsigned>(JSC::PropertyAttribute::Function), NoIntrinsic, { HashTableValue::NativeFunctionType, jsStringDecoderPrototypeFunction_end, 1 } },
+      };
+
+void JSStringDecoderPrototype::finishCreation(VM& vm, JSC::JSGlobalObject* globalThis)
+{
+    Base::finishCreation(vm);
+    reifyStaticProperties(vm, JSStringDecoder::info(), JSStringDecoderPrototypeTableValues, *this);
+}
+
+const ClassInfo JSStringDecoderPrototype::s_info = { "StringDecoder"_s, nullptr, nullptr, nullptr, CREATE_METHOD_TABLE(JSStringDecoderPrototype) };
+
+EncodedJSValue constructJSStringDecoder(JSGlobalObject* globalObject, CallFrame* callFrame)
+{
+    JSC::VM& vm = globalObject->vm();
+    auto encoding = BufferEncodingType::utf8;
+    if (callFrame->argumentCount() > 0) {
+      auto encoding_ = callFrame->argument(0).toString(globalObject);
+      std::optional<BufferEncodingType> opt = parseEnumeration<BufferEncodingType>(*globalObject, encoding_);
+      if (opt.has_value()) {
+        encoding = opt.value();
+      }
+    }
+    JSStringDecoderPrototype* prototype = JSStringDecoderPrototype::create(
+        vm, globalObject, reinterpret_cast<Zig::GlobalObject*>(globalObject)->JSStringDecoderPrototypeStructure());
+    JSStringDecoder* stringDecoder = JSStringDecoder::create(
+        vm, globalObject, JSStringDecoder::createStructure(vm, globalObject, prototype), encoding);
```

Comment:
This should be the LazyProperty, not the prototype

Reply:
oh... ok...